assets/js/app.js

```diff
@@ -175,6 +193,29 @@ function App({ uniqueId, socket }) {
         localStorage.removeItem('uniqueId');
     }
 
+    /**
+     * Register timeout for refreshing token.
+     */
+    function setupTokenRefresh() {
+        const token = localStorage.getItem('token');
+        const expire = localStorage.getItem('expire') ?? 30;
+
+        if (tokenTimeout !== null) {
+            clearTimeout(tokenTimeout);
+        }
+
+        // Refresh the token an hour before expire.
+        const nextRefresh = Math.max(expire * 1000 - Date.now() - 60 * 60 * 1000, 30 * 1000);
```

Comment:
Could it be simplified by `const now = Math.floor(Date.now() / 1000)` and `* 1000` only once on the result to get `nextRefresh` in ms for the timeout.

Reply:
Fixed